metriken-exposition/src/snapshotter.rs

```diff
@@ -0,0 +1,135 @@
+use crate::*;
+
+/// Produces a snapshot of metric readings.
+pub struct Snapshotter {
+    counters: bool,
+    gauges: bool,
+    atomic_histograms: bool,
+    rwlock_histograms: bool,
```

Comment:
Is there a use case where we would want to expose atomic histograms but not rwlock histograms (or vice versa)?

Reply:
Yes. Maybe there's a better way to handle it, but in Rezolus, the AtomicHistograms are only for counter/gauge derived histograms (to provide insight into behavior on finer-grained timescales than external sampling). While the RwLockHistograms are exclusively used for BPF true histograms. I think that for the external collection of Rezolus, we don't want to expose the counter/gauge derived histograms for msgpack exposition.

Reply:
This seems like what you actually want is more of the ability to generically filter which metrics are used to form the snapshot instead of filtering by type?

What about making the snapshot builder take an optional filter function instead a bunch of options to filter by type?

Reply:
That's better. We now have generic name and kind filtering.